app/models/early_allocation.rb

```diff
@@ -117,4 +121,11 @@ def stage1_eligible?
   def all_stage2_false?
     STAGE2_PLAIN_BOOLEAN_FIELDS.map(&method(:public_send)).none?
   end
+
+  def record_outcome
+    return self.outcome = 'eligible' if eligible?
```

Comment:
Think this would be nicer as a simple 'if-else' statement rather than early returns. Also I think a comment would be handy to explain the reason for this field. As we are setting it programattically, the validation should probably be strict? if we overwrite before_save with guaranteed one of 3 values, maybe the validation can just go away? (especially if the last 'if' is just an else in record_outcome)